src/FormSerializationWriter.cs

```diff
@@ -162,18 +173,27 @@ public void WriteCollectionOfPrimitiveValues<T>(string? key, IEnumerable<T>? val
     {
         if(depth > 0) throw new InvalidOperationException("Form serialization does not support nested objects.");
         depth++;
-        if(value == null && !additionalValuesToMerge.Any(static x => x != null)) return;
+
+        if (value == null)
+        {
+            var enumerator = additionalValuesToMerge.GetEnumerator();
+            bool hasAdditionalValues = enumerator.MoveNext();
+            if (enumerator is IDisposable de) de.Dispose();
+            if (!hasAdditionalValues) return;
+        }
```

Comment:
Is this equivalent to the previous version? This looke like `Any()` without a predicate.

It might help readability to just define your own tiny LINQ with the obvious implementation and use that. E.g. https://github.com/dotnet/corert/blob/master/src/Common/src/Internal/LowLevelLinq/LowLevelEnumerable.cs

The problem with the real LINQ is that it's not implemented the way the "obvious" implementation would be. It's _a lot_ more complicated.